topdown/cache/cache.go

```diff
@@ -154,11 +154,14 @@ func NewInterQueryCache(config *Config) InterQueryCache {
 func NewInterQueryCacheWithContext(ctx context.Context, config *Config) InterQueryCache {
 	iqCache := newCache(config)
 	if iqCache.staleEntryEvictionTimePeriodSeconds() > 0 {
-		cleanupTicker := time.NewTicker(time.Duration(iqCache.staleEntryEvictionTimePeriodSeconds()) * time.Second)
 		go func() {
+			cleanupTicker := time.NewTicker(time.Duration(iqCache.staleEntryEvictionTimePeriodSeconds()) * time.Second)
 			for {
 				select {
 				case <-cleanupTicker.C:
+					// NOTE: We stop the ticker and create a new one here to ensure that applications
+					// get _at least_ staleEntryEvictionTimePeriodSeconds with the cache unlocked;
+					// see https://github.com/open-policy-agent/opa/pull/7188/files#r1855342998
```

Comment:
Instead of putting a link here, it would be helpful to add some text explaining the situation.

Reply:
The link is to the discussion in this PR; I'm happy to add _more_ detail here, but I thought that the link to the original was probably also useful background.  Would the following work?

```suggestion
					// for more background, see https://github.com/open-policy-agent/opa/pull/7188/files#r1855342998
```

Reply:
A single line summary of the convo with the link would be great!

Reply:
I think that's what the two lines above the link are -- a summary of the discussion, and then a link for more information.

Reply:
Ok, can you please update the commit message by adding a description in the body and we can get this in.

Reply:
Updated the commit message to cover this discussion as well as the fixing of the flakes in the test (which was the primary motivation).